### DISCUSSION TITLE
link v-if on a page to result of click on q-item in q-itemlist in q-drawer layout

### DISCUSSION THREAD

**Discussion by D042657** [quasarframework/quasar#14234] [General - Components / Directives / etc]
Hi, 
I'm a complete newbe to UI programming. I'm trying to link the rendering of a component to the result of a click event on a q-item in a q-list that is contained in as a dynamic component in a q-drawer of the layout the page is linked to.

In the meta drawerContent of each route in routes.ts I specified what menue items should show up in the q-drawer.

        path: 'reason',
        component: () => import('pages/ReasonPage.vue'),
        meta: {
          breadCrumb: {
            icon: 'trip_origin',
            label: 'Reason',
            to: '/mladministration/reason',
          },
          **drawerContent:** [
            {
              label: 'search',
              icon: 'search',
            },
            {
              label: 'add',
              icon: 'add',
            },
            {
              label: 'update',
              icon: 'edit',
            },
            {
              label: 'merge',
              icon: 'merge',
            },
            {
              label: 'deprecate',
              icon: 'archive',
            },
            {
              label: 'delete',
              icon: 'delete',
            },

I created a component, that reads this meta.drawerContent and generates q-items populating a q-list in the drawer. When an item is clicked, it calls the clickedDrawerItemLink function to retunr/expose the label of the clicked item.

<template>
  <q-item class="items-center" v-for="option in **routeDrawerContent**" :key="option.label"
    clickable
    v-ripple
    :active="link === option.label"
    **@click="clickedDrawerItemLink(option.label)"**
    active-class="my-menu-link"
  >
    <q-item-section avatar>
      <q-icon :name=option.icon />
    </q-item-section>
    <q-item-label > {{ option.label }} </q-item-label>
  </q-item>
</template>

<script setup lang="ts">
import { ref, computed } from 'vue';
import { useRoute } from 'vue-router';

let link = ref('')

const routeDrawerContent = computed(() => {
  const route = useRoute();
  const drawerContent: {label: string, icon: string}[] = route.meta.drawerContent as {label: string, icon: string}[]
  return drawerContent;
});

const clickedDrawerItemLink = (link: string) => {
  **return link;**
};

</script>

In the layout, that is loaded and (as I see in the DOM) wrapping the page i placed this component generating the drawer content.

<template>
  <q-layout view="hHh Lpr fFf">

    <q-header elevated>
      <q-toolbar>
        <q-btn flat round dense icon="menu" @click="toggleLeftDrawer" />
        <bread-crumbs-component />
        <q-space />
        <q-btn dense flat icon="minimize" />
        <q-btn dense flat icon="crop_square" />
        <q-btn dense flat icon="close" />
      </q-toolbar>

      <dynamic-q-route-tabs-component />
    </q-header>

    <q-footer>
      <q-toolbar>
        <q-toolbar-title>
          Status/Info
        </q-toolbar-title>
      </q-toolbar>
    </q-footer>

    <q-drawer v-if="lenDrawerContent > 0" v-model="leftDrawerOpen" show-if-above side="left" bordered>
      <q-list bordered padding class="text-primary">
        **<dynamic-drawer-content/>**
      </q-list>
    </q-drawer>

    <q-page-container>
      <router-view />
    </q-page-container>
  </q-layout>
</template>

<script setup lang="ts">
import { ref, computed } from 'vue';
import { useRoute } from 'vue-router'
import BreadCrumbsComponent from '../components/BreadCrumbsComponent.vue';
import DynamicQRouteTabsComponent from '../components/DynamicQRouteTabsComponent.vue';
import DynamicDrawerContent from '../components/DynamicDrawerContent.vue';

let leftDrawerOpen = ref(false);

const toggleLeftDrawer = () => {
  leftDrawerOpen.value = !leftDrawerOpen.value;
};

const lenDrawerContent = computed((): number => {
  const route = useRoute();
  let drawerContent = route.meta.drawerContent as {label: string, icon:string}[];
  return drawerContent.length;
});
</script>

Finally I like a component ReasonAddComponent.vue to be rendered into the ReasonPage.vue if the q-item having label "add" was clicked.

<template>
  **<div v-if="clickedDrawerItemLink.value === 'add'">**
    <reason-add-component/>
  </div>
</template>

<script setup lang="ts">
import ReasonAddComponent from '../components/ReasonAddComponent.vue';
import **clickedDrawerItemLink** from '../components/DynamicDrawerContent.vue'

console.log(clickedDrawerItemLink)
</script>

Looks that I haven't understood yet, how to access/receive the click event of that q-item in the q-drawer of the wrapping q-layout on a page.

I'm grateful for any advice.

**Comment by metalsadman** [comment]
make a reproduction of that here https://stackblitz.com/edit/quasarframework-s8bzam
